OpenSim/Common/Component.cpp

```diff
@@ -615,86 +615,27 @@ void Component::setParent(const Component& parent)
 
 std::string Component::getFullPathName() const
```

Comment:
I like getting Full (or Absolute) and Relative Paths and `ComponentPath`s. I think we should transition to use `ComponentPath` instead of `std::string`, but we need to de/serialize ComponenPaths to do that effectively.  Keeping string variants with get<...>Name() may not be necessary if there is `ComponentPath::toString()`. I think we can discuss that as a subsequent improvement.